scylla-server-rust/src/processors/mqtt_processor.rs

```diff
@@ -161,14 +164,23 @@ impl MqttProcessor {
     /// returns the ClientData, or the Err of something that can be debug printed
     #[instrument(skip(self), level = Level::TRACE)]
     fn parse_msg(&self, msg: Publish) -> Result<ClientData, impl fmt::Debug> {
-        let data = serverdata::ServerData::parse_from_bytes(&msg.payload)
-            .map_err(|f| format!("Could not parse message topic:{:?} error: {}", msg.topic, f))?;
+        let topic = std::str::from_utf8(&msg.topic)
+            .map_err(|f| format!("Could not parse topic: {}, topic: {:?}", f, msg.topic))?;
+
+        // ignore command messages, less confusing in logs than just failing to decode protobuf
+        if topic.starts_with(CALYPSO_BIDIR_CMD_PREFIX) {
+            return Err(format!("Skipping command message: {}", topic));
```

Comment:
wait I'm confused, why would scylla ever get a command message?

Reply:
Because the MQTT server sends to everyone subscribed to the topic the command.  And scylla is subscribed to every topic so it can store them in it's database.

Reply:
Ahhh ok, makes sense